src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -27,6 +27,15 @@ public MockDirectoryInfo(IMockFileDataAccessor mockFileDataAccessor, string dire
             this.mockFileDataAccessor = mockFileDataAccessor ?? throw new ArgumentNullException(nameof(mockFileDataAccessor));
 
             originalPath = directoryPath;
+
+            if (directoryPath== null)
+            {
+                throw new ArgumentNullException("path", StringResources.Manager.GetString("VALUE_CANNOT_BE_NULL"));
+            }
+            if (directoryPath.Trim() == string.Empty)
+            {
+                throw CommonExceptions.PathIsNotOfALegalForm("path");
+            }
```

Comment:
Should this (similar to the proposed change in MockFileInfo) also rely on `mockFileSystem.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");` to check for a valid path?
```suggestion
            if (directoryPath== null)
            {
                throw new ArgumentNullException("path", StringResources.Manager.GetString("VALUE_CANNOT_BE_NULL"));
            }
            if (directoryPath.Trim() == string.Empty)
            {
                throw CommonExceptions.PathIsNotOfALegalForm("path");
            }
            
            originalPath = directoryPath;
```

Reply:
@siprbaum :
> Should this (similar to the proposed change in MockFileInfo) also rely on `mockFileSystem.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");` to check for a valid path?

To answer your question: I started implementing it using `PathVerifier.IsLegalAbsoluteOrRelative`, but it resulted in 47 failed tests; most often about "Illegal characters in path.".
I didn't have time to analyse them in depth to see, if I could refactor the constructor and/or the tests, to find a better solution. This was the change with the compromise with the least impact and I think changing more would need to be part of a different pull request (if at all).